lib/remoteproc/elf_loader.c

```diff
@@ -429,13 +429,13 @@ int elf_load_header(const void *img_data, size_t offset, size_t len,
 	}
 	metal_assert(*img_info != NULL);
 	load_state = elf_load_state(*img_info);
-	if (last_load_state != (int)*load_state)
+	if (last_load_state != *load_state)
 		return -RPROC_EINVAL;
 	/* Get ELF program headers */
 	if (*load_state == ELF_STATE_WAIT_FOR_PHDRS) {
 		size_t phdrs_size;
 		size_t phdrs_offset;
-		char **phdrs;
+		void **phdrs;
```

Comment:
void *phdrs;

Reply:
Can't change to void *, since we need save the allocated memory:
*phdrs= metal_allocate_memory(phdrs_size);